src/main/java/oracle/kubernetes/operator/rest/RestBackendImpl.java

```diff
@@ -474,4 +476,54 @@ private ResourceBundle getResourceBundle() {
     }
     throw new AssertionError(formatMessage(MessageKeys.RESOURCE_BUNDLE_NOT_FOUND));
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void upgradeApplications(String domainUID, UpgradeApplicationsModel appsToUpgrade ) {
+    LOGGER.entering(domainUID, "upgrade applications");
+
+    ClientHolder client = null;
+    try {
+      client = clientHelper.take();
+      upgradeApplications(client, domainUID, appsToUpgrade);
+    } finally {
+      recycleClient(clientHelper, client);
+    }
+    LOGGER.exiting();
+
+  }
+
+  private void upgradeApplications(ClientHolder client, String domainUID, UpgradeApplicationsModel appsToUpgrade) {
+    authorize(client, domainUID, Operation.update);
+
+    List<Domain> domains = getDomainsList(client);
```

Comment:
This seems inefficient.  Why list all domains just to get the single one with domainUID?

Reply:
These are pre-existing methods that I am reusing. It seems CallBuilder only has a listDomains(ns) method.